.github/actions/gen-docs/action.yaml

```diff
@@ -0,0 +1,56 @@
+name: "Generate all files related to documentation"
+
+description: "Generates all files related to documentation."
+
+inputs:
+  username:
+    description: "GitHub username"
+    required: true
+  email:
+    description: "GitHub email"
+    required: true
+  token:
+    description: "GitHub Token (must be a PAT for repository dispatch)"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install python and set up Poetry
+      uses: bakdata/ci-templates/actions/python-setup-poetry@v1.5.3
+      with:
+        poetry-version: "1.3.2"
+        python-version: "3.10"
+
+    - name: Install optional dependencies
+      shell: bash
+      run: poetry install --with docs
+
+    - name: Configure git
+      shell: bash
+      env:
+        USERNAME: ${{ inputs.username }}
+        EMAIL: ${{ inputs.email }}
+        TOKEN: ${{ inputs.token }}
+      run: |
+        git config --local user.name ${USERNAME}
+        git config --local user.email ${EMAIL}
+        git config --local user.password ${TOKEN}
```

Comment:
You can replace all of this with this action:
https://github.com/bakdata/ci-templates/blob/v1.17.0/actions/commit-and-push/action.yaml